datatypes/src/raster/tiling.rs

```diff
@@ -0,0 +1,206 @@
+            tile_pixel_size: tiling_specification.tile_size,
+            geo_transform: GeoTransform::new(
+                tiling_specification.origin_coordinate,
+                x_pixel_size,
+                y_pixel_size,
+            ),
+        }
+    }
+
+    pub fn upper_left_pixel_idx(&self, bounding_box: BoundingBox2D) -> GridIdx2D {
+        self.geo_transform
+            .coordinate_to_grid_idx_2d(bounding_box.upper_left())
+    }
+
+    pub fn lower_right_pixel_idx(&self, bounding_box: BoundingBox2D) -> GridIdx2D {
+        let lr_idx = self
+            .geo_transform
+            .coordinate_to_grid_idx_2d(bounding_box.lower_right());
+
+        lr_idx - 1
+    }
+
+    pub fn pixel_idx_to_tile_idx(&self, pixel_idx: GridIdx2D) -> GridIdx2D {
+        let GridIdx([y_pixel_idx, x_pixel_idx]) = pixel_idx;
+        let [y_tile_size, x_tile_size] = self.tile_pixel_size.into_inner();
+        let y_tile_idx = (y_pixel_idx as f32 / y_tile_size as f32).floor() as isize;
+        let x_tile_idx = (x_pixel_idx as f32 / x_tile_size as f32).floor() as isize;
+        [y_tile_idx, x_tile_idx].into()
+    }
+
+    pub fn pixel_idx_to_next_tile_idx(&self, pixel_idx: GridIdx2D) -> GridIdx2D {
+        let GridIdx([y_pixel_idx, x_pixel_idx]) = pixel_idx;
+        let [y_tile_size, x_tile_size] = self.tile_pixel_size.into_inner();
+        let y_tile_idx = (y_pixel_idx as f32 / y_tile_size as f32).ceil() as isize;
+        let x_tile_idx = (x_pixel_idx as f32 / x_tile_size as f32).ceil() as isize;
+        [y_tile_idx, x_tile_idx].into()
+    }
+
+    pub fn pixel_grid_box(&self, bounding_box: BoundingBox2D) -> GridBoundingBox2D {
+        let start = self.upper_left_pixel_idx(bounding_box);
+        let end = self.lower_right_pixel_idx(bounding_box);
+        GridBoundingBox2D::new_unchecked(start, end)
+    }
+
+    pub fn tile_grid_box(&self, bounding_box: BoundingBox2D) -> GridBoundingBox2D {
+        let start = self.pixel_idx_to_tile_idx(self.upper_left_pixel_idx(bounding_box));
+        let end = self.pixel_idx_to_tile_idx(self.lower_right_pixel_idx(bounding_box));
+        GridBoundingBox2D::new_unchecked(start, end)
+    }
+
+    /// generates the tile idx for the tiles intersecting the bounding box
+    pub fn tile_idx_iterator(
+        &self,
+        bounding_box: BoundingBox2D,
+    ) -> impl Iterator<Item = GridIdx2D> {
+        let GridIdx([upper_left_tile_y, upper_left_tile_x]) =
+            self.pixel_idx_to_tile_idx(self.upper_left_pixel_idx(bounding_box));
+
+        let GridIdx([lower_right_tile_y, lower_right_tile_x]) =
+            self.pixel_idx_to_tile_idx(self.lower_right_pixel_idx(bounding_box));
+
+        let y_range = upper_left_tile_y..=lower_right_tile_y;
+        let x_range = upper_left_tile_x..=lower_right_tile_x;
+
+        y_range.flat_map(move |y_tile| x_range.clone().map(move |x_tile| [y_tile, x_tile].into()))
+    }
+
+    /// generates the tile idx for the tiles intersecting the bounding box
```

Comment:
generates the tile INFORMATION. please also indicate the order in which the are created

Reply:
0429c17  
1c397a3